makepki.py

```diff
@@ -0,0 +1,362 @@
+    crl_builder = crl_builder.last_update(now)
+    crl_builder = crl_builder.next_update(now + DAY)
+
+    crl_filename = os.path.join(DIR, ca.nick + '.crl')
+
+    try:
+        f = open(crl_filename, 'rb')
+    except FileNotFoundError:
+        pass
+    else:
+        with f:
+            crl_pem = f.read()
+
+        crl = x509.load_pem_x509_crl(crl_pem, default_backend())
+
+        for revoked_cert in crl:
+            crl_builder = crl_builder.add_revoked_certificate(revoked_cert)
+
+    builder = x509.RevokedCertificateBuilder()
+    builder = builder.serial_number(serial)
+    builder = builder.revocation_date(now)
+
+    revoked_cert = builder.build(default_backend())
+
+    crl_builder = crl_builder.add_revoked_certificate(revoked_cert)
+
+    crl = crl_builder.sign(
+        private_key=ca.key,
+        algorithm=hashes.SHA256(),
+        backend=default_backend(),
+    )
+
+    crl_pem = crl.public_bytes(serialization.Encoding.PEM)
+
+    with open(crl_filename, 'wb') as f:
+        f.write(crl_pem)
+
+
+def gen_server_certs(nick_base, hostname, org, ca=None):
+    gen_cert(profile_server, nick_base, x509.Name([x509.NameAttribute(NameOID.ORGANIZATION_NAME, org), x509.NameAttribute(NameOID.COMMON_NAME, hostname)]), ca)
+    gen_cert(profile_server, nick_base + '-badname', x509.Name([x509.NameAttribute(NameOID.ORGANIZATION_NAME, org), x509.NameAttribute(NameOID.COMMON_NAME, 'not-' + hostname)]), ca)
+    gen_cert(profile_server, nick_base + '-altname', x509.Name([x509.NameAttribute(NameOID.ORGANIZATION_NAME, org), x509.NameAttribute(NameOID.COMMON_NAME, 'alt-' + hostname)]), ca, dns_name=hostname)
+    gen_cert(profile_server, nick_base + '-expired', x509.Name([x509.NameAttribute(NameOID.ORGANIZATION_NAME, org), x509.NameAttribute(NameOID.ORGANIZATIONAL_UNIT_NAME, 'Expired'), x509.NameAttribute(NameOID.COMMON_NAME, hostname)]), ca, warp=-2 * YEAR)
+    gen_cert(profile_server, nick_base + '-badusage', x509.Name([x509.NameAttribute(NameOID.ORGANIZATION_NAME, org), x509.NameAttribute(NameOID.ORGANIZATIONAL_UNIT_NAME, 'Bad Usage'), x509.NameAttribute(NameOID.COMMON_NAME, hostname)]), ca)
+    revoked = gen_cert(profile_server, nick_base + '-revoked', x509.Name([x509.NameAttribute(NameOID.ORGANIZATION_NAME, org), x509.NameAttribute(NameOID.ORGANIZATIONAL_UNIT_NAME, 'Revoked'), x509.NameAttribute(NameOID.COMMON_NAME, hostname)]), ca)
+    revoke_cert(ca, revoked.cert.serial_number)
+
+
+def gen_kdc_certs(nick_base, hostname, org, ca=None):
+    gen_cert(profile_kdc, nick_base + '-kdc', x509.Name([x509.NameAttribute(NameOID.ORGANIZATION_NAME, org), x509.NameAttribute(NameOID.ORGANIZATIONAL_UNIT_NAME, 'KDC'), x509.NameAttribute(NameOID.COMMON_NAME, hostname)]), ca)
+    gen_cert(profile_kdc, nick_base + '-badname-kdc', x509.Name([x509.NameAttribute(NameOID.ORGANIZATION_NAME, org), x509.NameAttribute(NameOID.ORGANIZATIONAL_UNIT_NAME, 'KDC'), x509.NameAttribute(NameOID.COMMON_NAME, 'not-' + hostname)]), ca)
```

Comment:
Please, just switch the order in naming, having `-kdc` first, e.g. `nick_base '-kdc-badname', this keeps the certificates together in `ls` output.